maintainers/scripts/haskell/upload-nixos-package-list-to-hackage.sh

```diff
@@ -0,0 +1,7 @@
+#! /usr/bin/env nix-shell
+#! nix-shell -i bash -p nix curl gnused -I nixpkgs=.
+
+package_list="$(nix-build -A haskell.package-list)/nixos-hackage-packages.csv"
+username=$(grep "^username:" ~/.cabal/config | sed "s/^username: //")
+password_command=$(grep "^password-command:" ~/.cabal/config | sed "s/^password-command: //")
```

Comment:
Three things:

- Could you add a short comment what this script does to the top?

-   Also, for these `username` and `password_command` lines, could you add a quick explanation of how they should be set?  Or maybe just linking to the Cabal manual?

    On my own system I just checked this file, but it looks like I don't have anything set for `username` or `password_command`:

    ```console
    $ cat ~/.cabal/config | grep username
    -- username:
    $ cat ~/.cabal/config | grep password-command
    -- password-command:
    ```

    I imagine this is probably because I use `stack` to upload everything to Hackage.

-   Could you add some documentation about running this script to `HACKING.md`?  I'm guessing we should be running this script every time we merge `haskell-updates` into `master`?

---------------------------------------

Other than the above, this LGTM.

Reply:
I will document all of this.

I just want to remark that I am not dead set on this way to do this. We can always also read the credentials from env variables or prompt for them or something like that. But it works nice for me and I didn‘t want to prematurely increase complexity.

Reply:
I have updated the documentation.

Let me know if anything is unclear.